### ISSUE TITLE
listFiles response type is not valid for folders

### ISSUE THREAD

**Issue by lorenzocestaro** [imagekit-developer/imagekit-nodejs#105]
The response type for the `listFIiles` API is only valid for file objects, but if the API is used to fetch folders the response will be different, for example:
```typescript
{
  type: 'folder',
  name: 'folder-name',
  createdAt: '2024-12-02T20:54:59.433Z',
  updatedAt: '2024-12-02T20:54:59.433Z',
  folderId: '674e1ea3e375273f609c',
  folderPath: '/path/to/folder'
}
```

Unfortunately this is not even documented in the [official API documentation](https://imagekit.io/docs/api-reference/digital-asset-management-dam/list-and-search-assets#List-and-search-assets) of ImageKit. And there might even be a third response type for `file-version`. All available types according to the docs are:
- `file`
- `file-version`
- `folder`
- `all`

From the top of my head a solution to this would be making the response type dependent on the input type parameter (with `all` returning a union of all types). This should be quite clean and users can easily narrow down type using `type` in each object.

**Comment by imagekitio** [comment]
Thanks for pointing out, we will fix types and improve docs soon.

**Comment by ahnv** [comment]
Hey @lorenzocestaro, this should be fixed in [version 6.0.0](https://github.com/imagekit-developer/imagekit-nodejs/releases/tag/6.0.0)
